tljh/config.py

```diff
@@ -291,18 +295,22 @@ def main(argv=None):
 
     args = argparser.parse_args(argv)
 
-    if args.action == 'show':
-        show_config(args.config_path)
-    elif args.action == 'set':
-        set_config_value(args.config_path, args.key_path, parse_value(args.value))
-    elif args.action == 'add-item':
-        add_config_value(args.config_path, args.key_path, parse_value(args.value))
-    elif args.action == 'remove-item':
-        remove_config_value(args.config_path, args.key_path, parse_value(args.value))
-    elif args.action == 'reload':
-        reload_component(args.component)
-    else:
-        argparser.print_help()
+    try:
+        if args.action == 'show':
+            show_config(args.config_path)
+        elif args.action == 'set':
+            set_config_value(args.config_path, args.key_path, parse_value(args.value))
+        elif args.action == 'add-item':
+            add_config_value(args.config_path, args.key_path, parse_value(args.value))
+        elif args.action == 'remove-item':
+            remove_config_value(args.config_path, args.key_path, parse_value(args.value))
+        elif args.action == 'reload':
+            reload_component(args.component)
+        else:
+            argparser.print_help()
+    except Exception as e:
```

Comment:
Rather than catching all errors here, let's try proactively checking `if os.geteuid() != 0`, which is the condition when it is run as a non-root user. Maybe show an early warning in this case, indicating that it may fail with permission errors?